src/main/java/de/tum/in/www1/artemis/service/competency/KnowledgeAreaService.java

```diff
@@ -51,12 +46,12 @@ public KnowledgeArea createKnowledgeArea(KnowledgeAreaDTO knowledgeArea) {
     /**
      * Updates an existing knowledge area with the provided data
      *
-     * @param knowledgeArea the new knowledge area values
+     * @param knowledgeArea   the new knowledge area values
+     * @param knowledgeAreaId the id of the knowledge area to update
      * @return the updated knowledge area
      */
-    public KnowledgeArea updateKnowledgeArea(KnowledgeAreaDTO knowledgeArea) {
-        knowledgeAreaIsValidOrElseThrow(knowledgeArea);
-        var existingKnowledgeArea = knowledgeAreaRepository.findByIdElseThrow(knowledgeArea.id());
+    public KnowledgeArea updateKnowledgeArea(KnowledgeAreaDTO knowledgeArea, long knowledgeAreaId) {
```

Comment:
```suggestion
    public KnowledgeArea updateKnowledgeArea(long knowledgeAreaId, KnowledgeAreaDTO knowledgeArea) {
```

If you let me nitpick xD
We use the ID first